File: android/src/main/kotlin/cl/ceisufro/fluttermbtilesextractor/Tile.kt
package cl.ceisufro.fluttermbtilesextractor

class Tile(val zoom: Int, val column: Int, val row: Int) {

    companion object {
        fun fromMap(map: Map<String, Any>): Tile {
            val zoom = map["zoom"] as Int
            val column = map["column"] as Int
            val row = map["row"] as Int
            return Tile(zoom, column, row)
        }

        fun toMapList(list: ArrayList<Tile>): List<Map<String, Any>> {
            val result = arrayListOf<Map<String, Any>>()
            list.forEach {
                result.add(it.toMap())
            }
            return result
        }
    }

    fun toMap(): HashMap<String, Any> {
        val map = hashMapOf<String, Any>()
        map.put("zoom", zoom)
        map.put("column", column)
        map.put("row", row)
        return map
    }
}

File: ios/Classes/TIle.swift
//
//  TIle.swift
//  flutter_mbtiles_extractor
//
//  Created by CEISUFRO on 09-07-18.
//

import Foundation
class Tile: NSObject{
    let zoom:Int
    let column:Int
    let row:Int
    
    init(zoom:Int,column:Int, row:Int) {
        self.zoom=zoom
        self.column=column
        self.row=row
    }
    
    static func fromMap(map:NSDictionary) -> Tile {
        let zoom = map["zoom"] as! Int
        let column = map["column"] as! Int
        let row = map["row"] as! Int
        return Tile(zoom: zoom, column: column, row: row)
    }
    
    static func toMapList(list: [Tile]) -> [NSDictionary] {
        var result:[NSDictionary] = []
        list.forEach { element in
            result.append(element.toMap())
        }
        return result
    }
    
    func toMap() -> NSDictionary{
        return [
            "zoom":self.zoom,
            "column":self.column,
            "row":self.row,
        ]
    }
}
